### PR TITLE
chore: negative test for heartbeat typing

### PR DESCRIPTION
**Pull request by crusso**

### CODE REVIEW

File: test/fail/bad-heartbeat.mo [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+actor {
+
+  let raw_rand = (actor "aaaaa-aa" : actor { raw_rand : () -> async Blob }).raw_rand;
+
+  var count = 0;
+  var max = 10;
+
+  system func heartbeat() : () { // reject, should be async ()
+    if (count < max) {
```
Comment (ggreif) [inline test/fail/bad-heartbeat.mo:11]: remove this?

File: test/fail/bad-heartbeat.mo [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      count += 1;
+    }
+  };
+
+  public shared func go() : async () {
```
Comment (ggreif) [inline test/fail/bad-heartbeat.mo:16]: ditto

File: test/fail/bad-heartbeat.mo [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import Prim "mo:⛔";
+
+actor {
+
+  let raw_rand = (actor "aaaaa-aa" : actor { raw_rand : () -> async Blob }).raw_rand;
```
Comment (ggreif) [inline test/fail/bad-heartbeat.mo:5]: drop this line?

**Comment by crusso** [comment]
yeah, fair 'nuff I'll simplify. I just blindly turned a positive test into a negative one..

**Comment by ggreif** [comment]
> yeah, fair 'nuff I'll simplify. I just blindly turned a positive test into a negative one..

It kind-of shadows the interesting part. I can also edit if you want!

File: test/fail/bad-heartbeat.mo [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (ggreif) [inline test/fail/bad-heartbeat.mo:6]:
```suggestion
```

File: test/fail/bad-heartbeat.mo [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  var count = 0;
+  var max = 10;
+
+  system func heartbeat() : () { // reject, should be async ()
+    if (count < max) {
+      count += 1;
+    }
```
Comment (ggreif) [inline test/fail/bad-heartbeat.mo:13]:
```suggestion
```

File: test/fail/bad-heartbeat.mo [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  public shared func go() : async () {
+     var attempts = 0;
+     while (count < max) {
+       ignore await raw_rand(); // yield to scheduler
+       attempts += 1;
+       if (attempts >= 50 and count == 0)
+         throw Prim.error("he's dead Jim");
+     };
+     Prim.debugPrint(debug_show {count});
+  };
```
Comment (ggreif) [inline test/fail/bad-heartbeat.mo:25]:
```suggestion
```

File: test/fail/ok/bad-heartbeat.tc.ok [hunk 7]
```diff
@@ -0,0 +1,4 @@
+bad-heartbeat.mo:10.3-14.4: type error [M0127], system function heartbeat is declared with type
```
Comment (ggreif) [inline test/fail/ok/bad-heartbeat.tc.ok:1]:
```suggestion
bad-heartbeat.mo: 8.3-9.4: type error [M0127], system function heartbeat is declared with type
```

File: test/fail/ok/bad-heartbeat.tc.ok [hunk 8]
```diff
@@ -0,0 +1,4 @@
+bad-heartbeat.mo: 8.3-9.4: type error [M0127], system function heartbeat is declared with type
```
Comment (ggreif) [inline test/fail/ok/bad-heartbeat.tc.ok:1]:
```suggestion
bad-heartbeat.mo:8.3-9.4: type error [M0127], system function heartbeat is declared with type
```

File: test/fail/bad-heartbeat.mo [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import Prim "mo:⛔";
+
+actor {
+
+  var count = 0;
+  var max = 10;
+
```
Comment (ggreif) [inline test/fail/bad-heartbeat.mo:7]:
```suggestion
```

File: test/fail/ok/bad-heartbeat.tc.ok [hunk 10]
```diff
@@ -0,0 +1,4 @@
+bad-heartbeat.mo:8.3-9.4: type error [M0127], system function heartbeat is declared with type
```
Comment (ggreif) [inline test/fail/ok/bad-heartbeat.tc.ok:1]:
```suggestion
bad-heartbeat.mo:4.3-5.4: type error [M0127], system function heartbeat is declared with type
```

**Review by ggreif — APPROVED** [review]

**Pull request merged**